gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskTypeId.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * Identity for a Google Compute Engine disk type.
+ */
+public final class DiskTypeId extends ZoneResourceId {
+
+  static final Function<String, DiskTypeId> FROM_URL_FUNCTION = new Function<String, DiskTypeId>() {
+    @Override
+    public DiskTypeId apply(String pb) {
+      return DiskTypeId.fromUrl(pb);
+    }
+  };
+  static final Function<DiskTypeId, String> TO_URL_FUNCTION = new Function<DiskTypeId, String>() {
+    @Override
+    public String apply(DiskTypeId diskTypeId) {
+      return diskTypeId.toUrl();
+    }
+  };
+
+  static final String REGEX = ZoneResourceId.REGEX + "diskTypes/[^/]+";
```

Comment:
seems like this can be private (unless you were planning to use it in tests later?).
Also, you can compile it to a Pattern to make the matching slightly faster.